flax/core/scope.py

```diff
@@ -619,8 +619,9 @@ def is_collection_empty(self, col: str) -> bool:
   def _mutable_collection(self, col: str) -> MutableCollection:
     """Returns the collection `col` as a mutable object."""
     assert self.is_mutable_collection(col), f'Collection {col} is not mutable'
+
     if col not in self._variables:
-      if self.parent:
+      if self.parent and self.name is not None:
```

Comment:
I think its safer to use an assert over `name` so the runtime behaviour remains the same:
```python
if self.parent:
  assert self.name is not None
  ...
```